src/xtgeo/grid3d/grid_properties.py

```diff
@@ -34,42 +116,175 @@
+            dataframe_dict["Y_UTMN"] = yc.get_active_npvalues1d()
+            dataframe_dict["Z_TVDSS"] = zc.get_active_npvalues1d()
+        else:
+            dataframe_dict["X_UTME"] = xc.values1d
+            dataframe_dict["Y_UTMN"] = yc.values1d
+            dataframe_dict["Z_TVDSS"] = zc.values1d
+
+    for prop in gridproperties:
+        if activeonly:
+            vector = prop.get_active_npvalues1d()
+        else:
+            vector = prop.values1d.copy()
+            # mask values not supported in Pandas:
+            if prop.isdiscrete:
+                vector = vector.filled(fill_value=0)
+            else:
+                vector = vector.filled(fill_value=np.nan)
+
+        if doubleformat:
+            vector = vector.astype(np.float64)
+        else:
+            vector = vector.astype(np.float32)
+
+        dataframe_dict[prop.name] = vector
+
+    return pd.DataFrame.from_dict(dataframe_dict)
+
+
 class GridProperties(_Grid3D):
     """Class for a collection of 3D grid props, belonging to the same grid topology.
 
+     It is a thin wrapper on a list that 1) checks that the GridProperties
+     belong to the same Grid (loosely). 2) Contains operations that can be
+     called on lists of GridProperty objects for easy discoverability.
+
+     Examples::
+         >>> import xtgeo
+         >>> # Create an
+         >>> grid_properties = xtgeo.GridProperties(props=[])
+         >>> # Get the dataframe via the gridproperties object
+         >>> grid_properties.get_dataframe()
+         Empty DataFrame...
+         >>> # Convert the gridproperties to a list
+         >>> grid_properties_list = list(grid_properties)
+         >>> # Get the dataframe of the list:
+         >>> gridproperties_dataframe(grid_properties_list)
+         Empty DataFrame...
+
+    Args:
+        ncol: Deprecated argument.
+        nrow: Deprecated argument.
+        nlay: Deprecated argument.
+        props: The list of GridProperty objects.
+
     See Also:
         The :class:`GridProperty` class.
     """
 
     def __init__(
         self,
-        ncol: Optional[int] = 4,
-        nrow: Optional[int] = 3,
-        nlay: Optional[int] = 5,
+        ncol: Optional[int] = None,
+        nrow: Optional[int] = None,
+        nlay: Optional[int] = None,
+        props: List[GridProperty] = None,
     ):
-        """Instantiating.
+        if ncol is not None:
+            warnings.warn("Initializing GridProperties with ncol is deprecated.")
+        else:
+            ncol = 4
+        if nrow is not None:
+            warnings.warn("Initializing GridProperties with nrow is deprecated.")
+        else:
+            nrow = 3
+        if nlay is not None:
+            warnings.warn("Initializing GridProperties with nrow is deprecated.")
+        else:
+            nrow = 5
 
-        Args:
-            ncol: Number of columns
-            nrow: Number of rows
-            nlay: Number of layers
+        if props:
```

Comment:
Warning/error if both props and ncol/nrow/nlay are set?

Reply:
The idea is that the user should not set `ncol/nrow/nlay` at all, so there will be a warning if that is set.

Reply:
I see that. My point is that if user do set ncol/row/lay, these values are completely ignored if props is also set, which may be confusing. I feel we should explicitly disallow giving both by raising an exception. This would not break backwards-comp since props-param dd not exist previously.

Reply:
Done :)